Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -508,7 +508,7 @@ FontCascade::CodePath FontCascade::codePath(const TextRun& run, std::optional<un
 #if USE(FONT_VARIANT_VIA_FEATURES) || USE(FREETYPE)
     // Because Font::applyTransforms() doesn't know which features to enable/disable in the simple code path, it can't properly handle feature or variant settings.
     // FIXME: https://bugs.webkit.org/show_bug.cgi?id=150791: @font-face features should also cause this to be complex.
-    if (m_fontDescription.featureSettings().size() > 0 || !m_fontDescription.variantSettings().isAllNormal())
+    if ((m_fontDescription.featureSettings().size() > 0 || !m_fontDescription.variantSettings().isAllNormal()) && !m_fontDescription.areLigaturesDisabled())
```

Comment:
I think we should either:
A) Delete this check entirely. `Font::applyTransforms()` _does_ know which features to enable/disable now.
B) Delete the complex text codepath entirely. This is a great opportunity to actually do it.

Reply:
How about we do (A) here, for performance. And I create a new bugzilla assigned to myself to do (B)?